src/main/java/net/starlark/java/eval/StringModule.java

```diff
@@ -80,6 +84,28 @@ static String slice(String s, int start, int stop, int step) {
     }
   }
 
+  // Practically the most strings in Starlark are ASCII strings.
+  // This is a cache of single-char strings to avoid allocation in s[i] operation.
+  // It is a nested class to delay the initialization of the cache until it is really needed.
```

Comment:
Is the allocation of an array of 128 strings really worth deferring through the classloader? A bigger array, or one less frequently needed, sure, but this one is tiny.

Reply:
It's CPU time I tried to save, but probably it's not a big deal. I'll remove the nested class.